File: chapters/03_session_and_dm_scope/v1_no_session.ts
// TypeScript mirror for Chapter 03: Session and DM Scope (v1)
// Run with: npx ts-node v1_no_session.ts

function handleWithoutSession(text: string): { session_key: string; reply: string } {
  return {
    session_key: "ephemeral",
    reply: `stateless:${text}`,
  };
}

function main(): void {
  console.log(handleWithoutSession("hello"));
}

main();

File: chapters/03_session_and_dm_scope/v1_no_session.py
from __future__ import annotations


def handle_without_session(text: str) -> dict[str, str]:
    return {
        "session_key": "ephemeral",
        "reply": f"stateless:{text}",
    }


def main() -> None:
    print(handle_without_session("hello"))


if __name__ == "__main__":
    main()
